src/app/organizations/manage/reset-password.component.ts

```diff
@@ -0,0 +1,196 @@
+                        encryptedPrivateKey = response.encryptedPrivateKey;
+                    } else {
+                        throw new Error('Reset Password Details response is null');
+                    }
+
+                    // Decrypt Organization's encrypted Private Key with org key
+                    const orgSymKey = await this.cryptoService.getOrgKey(this.organizationId);
+                    const decPrivateKey = await this.cryptoService.decryptToBytes(new EncString(encryptedPrivateKey), orgSymKey);
+
+                    // Decrypt User's Reset Password Key to get EncKey
+                    const decValue = await this.cryptoService.rsaDecrypt(resetPasswordKey, decPrivateKey);
+                    const userEncKey = new SymmetricCryptoKey(decValue);
+
+                    // Create new key and hash new password
+                    const newKey = await this.cryptoService.makeKey(this.newPassword, this.email.trim().toLowerCase(),
+                        kdfType, kdfIterations);
+                    const newPasswordHash = await this.cryptoService.hashPassword(this.newPassword, newKey);
+
+                    // Create new encKey for the User
+                    const newEncKey = await this.cryptoService.remakeEncKey(newKey, userEncKey);
+
+                    // Create request
+                    const request = new OrganizationUserResetPasswordRequest();
+                    request.key = newEncKey[1].encryptedString;
+                    request.newMasterPasswordHash = newPasswordHash;
+
+                    // Change user's password
+                    return this.apiService.putOrganizationUserResetPassword(this.organizationId, this.id, request);
+                });
+
+            await this.formPromise;
+            this.platformUtilsService.showToast('success', null, this.i18nService.t('resetPasswordSuccess'));
+            this.onPasswordReset.emit();
+        } catch { }
+    }
+
+    updatePasswordStrength() {
+        if (this.newPasswordStrengthTimeout != null) {
+            clearTimeout(this.newPasswordStrengthTimeout);
+        }
+        this.newPasswordStrengthTimeout = setTimeout(() => {
+            const strengthResult = this.passwordGenerationService.passwordStrength(this.newPassword,
+                this.getPasswordStrengthUserInput());
+            this.masterPasswordScore = strengthResult == null ? null : strengthResult.score;
+        }, 300);
+    }
+
+    private getPasswordStrengthUserInput() {
+        let userInput: string[] = [];
+        const atPosition = this.email.indexOf('@');
+        if (atPosition > -1) {
+            userInput = userInput.concat(this.email.substr(0, atPosition).trim().toLowerCase().split(/[^A-Za-z0-9]/));
+        }
+        if (this.name != null && this.name !== '') {
+            userInput = userInput.concat(this.name.trim().toLowerCase().split(' '));
+        }
+        return userInput;
+    }
```

Comment:
Exists in 2 components (3 now), probably a good time to extract it shared code to ensure consistent behavior`.